### ISSUE TITLE
ESP32 interaction with Ultrasonic Sensor

### ISSUE THREAD

**Issue by Hendini21** [Hendini21/Local_Flood_Navigation#2]
Hey Hieu, 
It has been a while. Happy New Year. I hope you are enjoying the holidays. 
I am having a problem with my ESP32 and the ultrasonic sensor. I will send you pictures and the code file. 
Thanks again. 
-Alex

**Comment by Hendini21** [comment]
@
![image](https://user-images.githubusercontent.com/50623455/71647519-e3c4fb80-2cbd-11ea-8ee3-05d29ce4de06.jpg)
![image](https://user-images.githubusercontent.com/50623455/71647528-02c38d80-2cbe-11ea-9d8d-a0d54a8f65e8.jpg)
![image](https://user-images.githubusercontent.com/50623455/71647533-153dc700-2cbe-11ea-9b7c-cdd30c1d320f.jpg)
![image](https://user-images.githubusercontent.com/50623455/71647560-9432ff80-2cbe-11ea-9244-612c23daa49a.jpg)

**Comment by Hendini21** [comment]
`const int trigPin = 4;
const int echoPin = 5; 
long duration;
int S;
int inch;
void setup() {
pinMode(trigPin, OUTPUT); // Sets the trigPin as an Output
pinMode(echoPin, INPUT); // Sets the echoPin as an Input
Serial.begin(9600); // Starts the serial communication
}

void loop() {
// Clears the trigPin
digitalWrite(trigPin, LOW);
delayMicroseconds(2);

digitalWrite(trigPin, HIGH);
delayMicroseconds(10);
digitalWrite(trigPin, LOW);

// Reads the echoPin, returns the sound wave travel time in microseconds
duration = pulseIn(echoPin, HIGH);



S= duration*0.034/2;         //Calculating the distance traveled in cm
inch = duration*0.0133/2;    // Calculating the distance in inches


// Prints the distance on the Serial Monitor
Serial.print("Distance: ");
Serial.println(S);
Serial.print("cm");
delay(10);
Serial.print("Distance: ");
Serial.println(inch);
Serial.print("in");
}
`

**Comment by cheehieu** [comment]
Hey Alex. Can you elaborate on what the problem is? Are you getting any readings from the sensor?

**Comment by Hendini21** [comment]
No, the readings aren't taking place, nothing is happening at all.

**Comment by Hendini21** [comment]
That's the problem this is the standard code for using this kind of ultrasonic sensor, so it should work, but it doesn't work.

**Comment by cheehieu** [comment]
It's a bit difficult for me to debug without seeing the hardware in person... But I would make sure to check the following:

- [x] Confirm everything is wired up correctly
- [x] Confirm sensor is powered sufficiently
- [x] Where did you get the example code from? Is there a tutorial?
- [ ] Is the sensor exactly the same as the one used in the example code?
- [ ] Confirm the code is running properly
- [x] Confirm baud rate is correct for the serial output to display the readings

**Comment by Hendini21** [comment]
The sensor is the same is almost every aspect, the only difference is that it is built by someone else. It runs the exact same way though.

**Comment by cheehieu** [comment]
Maybe it is a faulty sensor? Can you try running your code with another one to see if you get the same results?

**Comment by Hendini21** [comment]
The sensor is brand new. I got it the day I tried running the code and it didn’t work. I don’t know what’s wrong with my layout if there is anything wrong.

**Comment by Hendini21** [comment]
But yes, I will try the code with another sensor.

**Comment by Hendini21** [comment]
I got another sensor and am trying to figure out if I just have my board set incorrectly, but I don't know if I do.

**Comment by Hendini21** [comment]
Nevermind, I got it working.
